src/references/script_debugger.md

```diff
@@ -0,0 +1,100 @@
+{: .image-inline-text .image-background }
+
+## Adding and Removing Breakpoints
+
+A breakpoint is a marker that can set for a specific line in a script. More than one marker can be set in a script, but only one can be set for each line. Markers can be set by left clicking with the mouse in the far left margin of the **Script Editor**.
+
+### Adding a Breakpoint
+
+A breakpoint can be added so the **Script Debugger** knows where to pause execution. Adding a breakpoint can be done by left clicking with the mouse in the far left margin of the **Script Editor**. On clicking in the margin, a red circle will be added to the margin, indicating that line has a breakpoint.
+
+![!Add Breakpoint](../img/ScriptDebugger/add_breakpoint.png){: .center loading="lazy" }
+
+### Removing a Breakpoint
+
+Removing a breakpoint works the same as adding a breakpoint. Clicking on the red circle in the far left margin of **Script Editor** will remove the breakpoint. When the game is played, the **Script Debugger** will not pause execution if no breakpoints are present.
+
+## Stepping Through a Script
+
+Stepping through a script is an important part of debugging, it allows creators to walk through each line of the script to inspect the variables and objects to see what is contained in them.
+
+### Step Into
+
+**Step Into** ![Step Into](../img/ScriptDebugger/step_into_icon.png)will be frequently used to step through a script line by line. This also includes functions. If a function is about to be called, and the code needs to be debugged in that function, then the next step is to go into that function and debug it line by line by using **Step Into**.
+{: .image-inline-text .image-background }
+
+### Step Over
+
+**Step Over** ![Step Over](../img/ScriptDebugger/step_over_icon.png)will execute as one complete step. For example, if the current line is a function, then stepping over that function will move to the next line instead of stepping through each line of that function.
+{: .image-inline-text .image-background }
+
+### Step Out
+
+**Step Out** ![Step Out](../img/ScriptDebugger/step_out_icon.png)is used to step out of a function. For example, when only part of a function needs to be debugged, stepping out will tell the **Script Debugger** to run the rest of that function.
+{: .image-inline-text .image-background }
+
+## Select a Script Task
+
+When a project gets bigger, more scripts are created to handle different parts of the game. Breakpoints can be added to different scripts, and the **Script Debugger** will pause execution for whichever script is running first that has a breakpoint set. Scripts that have not been completed can be stepped through. This can be done from **Script Task** drop down in the **Script Debugger** window.
```

Comment:
Cut the first sentence, and reframe this paragraph to indicate that this is about the behavior of multiple scripts and threads ...

How close does Script Task match Task in the way it is used to designate separate execution threads?

Reply:
Script tasks run 1 by 1, so while they are separate "threads", they don't run parallel to each other from what I see based on the way the debugger lists them in the list. Also data from completed tasks is lost which is a shame.